internal/kafka/internal/workers/kafka_mgrs/promotion/internal/actions/update_kafka_request_action.go

```diff
@@ -0,0 +1,37 @@
+package actions
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/workers/kafka_mgrs/promotion/chain"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+)
+
+var _ chain.ReconcileAction[PromotionContext] = &UpdateKafkaRequestAction{}
+
+type UpdateKafkaRequestAction struct {
+	kafkaService services.KafkaService
+}
+
+func NewUpdateKafkaRequestAction(kafkaService services.KafkaService) chain.ReconcileAction[PromotionContext] {
+	return &UpdateKafkaRequestAction{
+		kafkaService: kafkaService,
+	}
+}
+
+func (u UpdateKafkaRequestAction) PerformJob(kafkaRequest *dbapi.KafkaRequest, currentResult chain.Result[PromotionContext]) (chain.Result[PromotionContext], bool, error) {
+	// gorm ignores zero values, so to zero `PromotionStatus` and `PromotionDetails` we need to use a map
+	updates := map[string]any{}
+	updates["PromotionStatus"] = ""
```

Comment:
This does not seem accurate.

The fields in the DB are `promotion_status`,  `promotion_details`, `actual_kafka_billing_model` and `subscription_id`. This also makes me think we need an integration test for this to verify it with a real DB.

Does gorm update it correctly with the current implementation? Maybe it is flexible enough to detect it but I think we should use the values in the DB as that's what the key is intended to be

Reply:
An integration test has already been added, and it looks like gorm correctly updates the fields.
I will, however, change this.